### PR TITLE
74 extract formcontrol component to promote code reusability

### PR DESCRIPTION
**Pull request by rhysmah**
## Summary
Attempted to break down the forms, `LoginForm` and `RegistrationForm`, into smaller pieces.

- Exported login controls, consisting of `Remember me?` and `Forgot password`, into a new file called `Login Controls`
- Exported validation schemas into a separate `ValidationSchemas` file
- Created a `TextInput` component. This uses `useFormat()` hook, which receives state, validation, and submission from `<Formik>`. We deconstruct the information received into two fields, `field`, and `meta`, which we spread across the appropriate pieces in `TextInput`.
- Use this `TextInput` field in `RegistrationForm` and `LoginForm`, respectively. 

## Testing

1. Navigate to the landing page using Deploy Preview (below)
2. Navigate to the `Login`
3. Click `Submit` and confirm that the email and password fields outline in red and display error messages
4. Fill in the email, don't fill in password, click `Submit`, and confirm the password field goes red and displays an error
5. Fill in the password, don't fill in email, click `Submit`, and confirm the email field goes red and displays an error
6. Fill in both fields with valid inputs and click the `Submit`; confirm, in Google Developer Tools, that an object is created containing both the entered email and password.

### CODE REVIEW

File: apps/frontend/src/App.tsx [hunk 1]
```diff
@@ -2,15 +2,18 @@ import React from 'react';
...
+
 import LoginPage from './pages/LoginPage';
 import LoginForm from './components/LoginForm';
 
 function App() {
   return (
     <Routes>
       <Route path="/" element={<LandingPage />} />
-      <Route path="/register" element={<Registration />} />
+      <Route path="/register" element={<Registration formComponent={<RegistrationForm />} />} />
```
Comment (toco-t) [inline apps/frontend/src/App.tsx:16]: Passing `formComponent` prop inside `App.tsx` probably isn't the best approach (i.e. this page shouldn't be aware of the lower-level implementation details of sign-up and login pages).

**Review by toco-t — CHANGES_REQUESTED** [review]
I've also noticed that the error message for the password section on the registration page doesn't seem to be right.

image

File: apps/frontend/src/App.tsx (hunk 1 above)
Comment (rhysmah) [reply apps/frontend/src/App.tsx]: Great point.

I modified the two components so they no longer take forms as props; instead, I simply instantiate the forms within the pages themselves.

Screenshot 2024-03-01 at 09 26 53
Screenshot 2024-03-01 at 09 26 19

File: apps/frontend/src/components/ValidationSchemas.tsx
Comment (toco-t) [inline apps/frontend/src/components/ValidationSchemas.tsx:1]: I feel the schemas should be placed somewhere else (e.g. apps/frontend/src/schemas/schemas.ts)...?

**Review by toco-t — APPROVED** [review]
LGTM!

**Pull request merged**